fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+    fetch('/api/todos').then(response => response.json()).then((todos) => {
+      this.setState({ todos });
+    })
+  }
+
+  addTodo(newTodo) {
+    fetch('/api/todos', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newTodo),
+    })
+      .then(response => response.json())
+      .then(todos => {
+        let newList = this.state.todos;
+        newList.push(todos);
+        this.setState({ newList });
+      }).catch(err => console.error(err));
+  }
+
+  toggleCompleted(todoId) {
+    let targetObject = null;
+
+    const findTarget = this.state.todos.map((todo) => {
```

Comment:
The problem here is that the object you are creating and updating here does not contain all of the proper information from the database. Make sure that you place the object which is returned from the database into the array at its original position.